blogposts/2020/introducing-codeintel-db.md

```diff
@@ -0,0 +1,65 @@
+2. We have good instructions and tips for backup and recovery of our Postgres data store, but have never officially recommended backing up other persistent volumes.
+3. We locked ourselves into a node topology which is not easily scalable. We essentially have a singleton service over a persistent volume. In order to scale this service horizontally (which would become necessary once either the amount of data or the frequency of queries increased), we would need to shard the data across multiple volumes. This throws a lot of wrenches into our plans: we need to be able to automatically move data on increase or decrease of shard counts; we need to be able to route requests to the correct node with that data; we need to ensure that reconciliation between the data on disk and the metadata we store in Postgres is aware that there are multiple nodes where data _could_ reside.
+4. There is no efficient way to implement certain data access patterns we may need in the future. For example, looking for data matching a certain identifier _name_ would require us to open every SQLite database on disk (and, no, _batching it_ is not a solution).
+
+Instead of forging ahead and tackling the complexity we've made for ourselves, we decided to take a step back and re-evaluate whether or not our previous architecture choices were meeting our current needs. And it seems like they did not.
+
+Moving this data into Postgres seems like an obvious choice that will (eventually) be able to reduce or completely remove all of the burdens listed above.
+
+1. Reconciling pieces of data in the same datastore is trivial compared to reconciling pieces of data in two different datastores, especially if it should be done transactionally. Even though we're not in the same _physical_ database instance, having all of our data in _some_ Postgres-compatible store opens up the possibility of transactions spanning multiple physical Postgres instances via [postgres_fdw](https://www.postgresql.org/docs/12/postgres-fdw.html#id-1.11.7.42.12).
+2. Moving data to an existing technology allows us to re-use the same advice and documentation for backup and disaster recovery strategies. Removing the number of knobs that must be turned by site-admins is always going to be a win across the board.
+3. Moving from an embedded database to a server-client database removes the nasty singleton property on the service. Now that multiple servers can interact with the same data concurrently without risk of corruption, we can scale the services horizontally without requiring additional effort into implementing sticky sessions or complex request routing rules.
+4. Moving data into a single relational store allows us to create additional indexed views into the data. We can now query over dimensions that were not possible before without opening tends of thousands of relatively small database files.
+
+## Notes for site admins
```

Comment:
Do we want this blog post to be the source of truth for information like this?

cc @christinaforney

Reply:
This information is already available in the upgrade docs and the changelog, I just wanted to call it out as an extra precaution.

Reply:
We already did have a question in a support channel about these values, so it's apparently pretty easy to glaze over some of the release notes.

Reply:
I think the right way to call it out is to link to a source of truth. If those documents don't have sufficient information (customers have asked questions) then I think we should add context to the source of truth, not add more information here in this blog post.

Reply:
There are already links to https://docs.sourcegraph.com/admin/external_database and https://docs.sourcegraph.com/admin/updates in this text.

As for the question we got already:

```
does this use the pgsql instance we configured?  it looks like it is trying to stand up another db for this.... if that is the case, i need to find the settings to configure it to use the HA pgsql instance we have placed
```

I think the addition just took them by surprise, because their intuition about how to update their deployment was correct.

Here is the text in the upgrading section:

```
This release introduces a second database instance, codeintel-db. If you have configured Sourcegraph with an external database, then update the CODEINTEL_PG* environment variables to point to a new external database as described in the external database documentation. Again, these must not point to the same database or the Sourcegraph instance will refuse to start.
```

Reply:
So my question is, why not delete all the content and just keep the links? If there is any content that you hesitate to delete, ask yourself if it belongs somewhere else (e.g. inside one of those links).

Reply:
The information under the data retention policy I think is useful, but would not fit nicely inside the changelog (where the parameter is introduced).

Maybe this isn't the best place for it, but I don't think the changelog is the best places for it either.

Reply:
Let me know if you need more input on this post (I skimmed through it, and my concerns are the same that Nick mentions above - making sure the context is in the appropriate location).

Reply:
I'm assuming Nick and I will discuss this online in our 1-1 tomorrow, but I'll tap you if we discuss it earlier in case you're interested.